extension/src/pipeline/quickPick.ts

```diff
@@ -61,109 +68,160 @@ const pickTemplateAndFields = async (
+  dataArr: UnknownValue[]
+): { arrLength: number; fields: string[] } | undefined => {
   const firstArrVal: UnknownValue = dataArr[0]
   if (!isValueTree(firstArrVal)) {
-    return []
+    return
   }
   const fieldObjKeys = Object.keys(firstArrVal)
   const objsHaveSameKeys = dataArr.every(
     val => isValueTree(val) && isEqual(fieldObjKeys, Object.keys(val))
   )
-  return objsHaveSameKeys ? fieldObjKeys : []
+  if (!objsHaveSameKeys) {
+    return
+  }
+  return { arrLength: dataArr.length, fields: fieldObjKeys }
 }
 
-const getFieldsFromValue = (data: UnknownValue): string[] => {
+const getMetricInfoFromValue = (
+  data: UnknownValue
+): { arrLength: number; fields: string[] } | undefined => {
   const isArray = Array.isArray(data)
   const isObj = isValueTree(data)
   if (!isArray && !isObj) {
-    return []
+    return
   }
 
   const maybeFieldsObjArr = isArray ? data : data[Object.keys(data)[0]]
 
-  return Array.isArray(maybeFieldsObjArr)
-    ? getFieldsFromArr(maybeFieldsObjArr)
-    : []
-}
+  if (!Array.isArray(maybeFieldsObjArr)) {
+    return
+  }
 
-const showNotEnoughKeysToast = () =>
-  Toast.showError(
-    'The requested file or files do not contain enough keys (columns) to generate a plot. Does the file or files follow the DVC plot guidelines for [JSON/YAML](https://dvc.org/doc/command-reference/plots/show#example-hierarchical-data) or [CSV/TSV](https://dvc.org/doc/command-reference/plots/show#example-tabular-data) files?'
-  )
+  return getMetricInfoFromArr(maybeFieldsObjArr)
+}
 
-const getFieldsFromDataFiles = (
-  dataArr: { data: UnknownValue; file: string }[]
-) => {
-  const keys: {
-    [file: string]: string[]
-  } = {}
-  let keysAmount = 0
+const dvcPlotGuidelinesQuestion =
+  'Does the file contain data and follow the DVC plot guidelines for [JSON/YAML](https://dvc.org/doc/command-reference/plots/show#example-hierarchical-data) or [CSV/TSV](https://dvc.org/doc/command-reference/plots/show#example-tabular-data) files?'
+
+const validateSingleFileData = ({
+  file,
+  data
+}: {
+  data: UnknownValue
+  file: string
+}) => {
+  const { fields = [] } = getMetricInfoFromValue(data) || {}
+
+  if (fields.length < 2) {
+    void Toast.showError(
+      `${file} does not contain enough keys (columns) to generate a plot. ${dvcPlotGuidelinesQuestion}`
+    )
+    return
+  }
 
-  for (const { file, data } of dataArr) {
-    const fields = getFieldsFromValue(data)
+  return [{ fields, file }]
+}
 
-    if (fields.length === 0) {
-      void showNotEnoughKeysToast()
+const validateMultiFilesData = (
+  filesData: { data: UnknownValue; file: string }[]
+) => {
+  const filesArrLength: Set<number> = new Set()
+  const keys: FileFields = []
+
+  for (const { file, data } of filesData) {
+    const metricInfo = getMetricInfoFromValue(data)
+    if (!metricInfo) {
+      void Toast.showError(
+        `${file} does not contain any keys (columns) to generate a plot. ${dvcPlotGuidelinesQuestion}`
```

Comment:
[C] I don't think you need to split out no keys from < 2 keys. `${file} does not contain enough keys (columns) to generate a plot. ${dvcPlotGuidelinesQuestion}` is enough for both situations.

Reply:
Done :)